bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/DecimalValue.java

```diff
@@ -90,6 +101,14 @@ public DecimalValue(String value, DecimalValueKind valueKind) {
         this.valueKind = valueKind;
     }
 
+    private static BigDecimal getValidDecimalValue(BigDecimal bd) {
```

Comment:
Shall we move the hex value check inside this method as well?
So that we can keep all the validations in one method.

Reply:
This method has been used in some other places where we need to pass `BigDecimal` value as an argument. We need to pass a hexadecimal string value to the method `hexToDecimalFloatingPointNumber`. So moving it here may need another method which needs to get BigDecimal as an argument. IMO, keeping as it is may be more convenient. WDYT?